src/hotspot/share/services/threadService.cpp

```diff
@@ -229,14 +229,14 @@ Handle ThreadService::get_current_contended_monitor(JavaThread* thread) {
   // inflated again and to be associated with a completely different
   // ObjectMonitor by the time this object reference is processed
   // by the caller.
-  ObjectMonitor *wait_obj = thread->current_waiting_monitor();
+  ObjectMonitor* volatile wait_obj = thread->current_waiting_monitor();
 
   oop obj = NULL;
   if (wait_obj != NULL) {
     // thread is doing an Object.wait() call
     obj = wait_obj->object();
   } else {
-    ObjectMonitor *enter_obj = thread->current_pending_monitor();
+    ObjectMonitor* volatile enter_obj = thread->current_pending_monitor();
```

Comment:
Please add a comment above this declaration. Something like:

```
// Using 'volatile' to prevent the compiler from generating code that
// reloads 'enter_obj' from memory when used below.